src/Mesh/DerivedData.C

```diff
@@ -801,4 +802,311 @@ genEsued( const std::vector< std::size_t >& inpoel,
+  Assert( nfpe > 0, "Attempt to call genNtfac() with zero faces per element" );
+
+  auto nelem = esuelTet.size()/nfpe;
+
+  std::size_t ntfac(0), nifac(0);
+  // note: nbfac is assumed to be already initialized
+
+  // loop through elements surrounding elements to find number of internal faces
+  for (std::size_t e=0; e<nelem; ++e)
+  {
+    for (std::size_t ip=nfpe*e; ip<nfpe*(e+1); ++ip)
+    {
+      if (esuelTet[ip] != -1)
+      {
+        if ( e<static_cast< std::size_t >(esuelTet[ip]) )
+        {
+          ++nifac;
+        }
+      }
+    }
+  }
+
+  ntfac = nifac + nbfac;
+
+  return ntfac;
+}
+
+std::vector< int >
+genEsuf( std::size_t nfpe,
+         const std::size_t& ntfac,
+         const std::size_t& nbfac,
+         const std::map< int, std::vector< std::size_t > >& belem,
+         const std::vector< int >& esuelTet )
+// *****************************************************************************
+//  Generate derived data, elements surrounding faces
+//  added by Aditya K Pandare
+//! \param[in] nfpe  Number of faces per element.
+//! \param[in] ntfac Total number of faces.
+//! \param[in] nbfac Number of boundary faces.
+//! \param[in] belem Boundary element map according to side-sets.
+//! \param[in] esuelTet Elements surrounding elements.
+//! \return Elements surrounding faces.
+//! \details The unsigned integer vector gives the IDs of the elements to the 
+//    left and the right of each face in the mesh. The convention followed 
+//    throughout is : The left element always has an ID smaller than the ID of 
+//    the right element.
+// *****************************************************************************
+{
+  Assert( esuelTet.size()%nfpe == 0, 
+                  "Size of esuelTet must be divisible by nfpe" );
+  Assert( nfpe > 0, "Attempt to call genEsuf() with zero faces per element" );
+
+  auto nelem = esuelTet.size()/nfpe;
+
+  std::vector< int > esuf(2*ntfac);
+
+  // counters for number of internal and boundary faces 
+  std::size_t icoun(2*nbfac), bcoun(0);
+
+  // loop to get face-element connectivity for internal faces
+  for (std::size_t e=0; e<nelem; ++e)
+  {
+    for (std::size_t ip=nfpe*e; ip<nfpe*(e+1); ++ip)
+    {
+      auto jelem = esuelTet[ip];
+      if (jelem != -1)
+      {
+        if ( e < static_cast< std::size_t >(jelem) )
+        {
+          esuf[icoun] = static_cast< int >(e);
+          esuf[icoun+1] = static_cast< int >(jelem);
+          icoun = icoun + 2;
+        }
+      }
+    }
+  }
+
+  bcoun = 0;
+  for (auto& iss : belem)
+  {
+    for (auto f : iss.second)
+    {
+      // define ielem from elems in side set connected to f!
+      auto ielem = f;
+      esuf[bcoun] = static_cast< int >(ielem);
+      esuf[bcoun+1] = -1;  // outside domain
+      bcoun = bcoun + 2;
+    }
+  }
+
+  return esuf;
+}
+
+std::vector< std::size_t >
+genInpofaTet( const std::size_t& ntfac,
+              const std::size_t& nbfac,
```

Comment:
Since these two are small, they could be just std::size_t, instead of const-ref. (I believe both will be 64 bits, so prefer shorter code.)